templates/elasticsearch.yml.j2

```diff
@@ -39,6 +42,10 @@ action.auto_create_index: {{ es_action_auto_create_index }}
 xpack.security.enabled: true
 {% endif %}
 
+{% if es_enable_xpack and es_api_basic_auth_username is defined and es_api_basic_auth_password is defined %}
+xpack.security.transport.ssl.enabled: true
```

Comment:
Could you explain why this needs to be hardcoded in? The [default setting is `false`](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html#transport-tls-ssl-settings) which would be true for anybody who currently has this deployed which would make this a backwards incompatible change.